lib/src/source_visitor.dart

```diff
@@ -2528,6 +2528,21 @@ class SourceVisitor extends ThrowingAstVisitor {
     token(node.superKeyword);
   }
 
+  @override
+  void visitSuperFormalParameter(SuperFormalParameter node) {
```

Comment:
This is virtually identical to the code for `visitFieldFormalParameter()` (`this.` parameters), which is to be expected. Unfortunately, I couldn't easily hoist it out to something reusable because FieldFormalParameter and SuperFormalParameter don't define their fields on any shared base class.